src/controllers/listener.ts

```diff
@@ -16,20 +16,20 @@ export async function setupListener({ nats, archipelago, config, logs }: Compone
   const peerExpirationInterval = setInterval(() => {
     const expiredHeartbeatTime = Date.now() - checkHeartbeatInterval
 
-    const inactivePeers = Array.from(lastPeerHeartbeats)
-      .filter(([_, lastHearbeat]) => lastHearbeat < expiredHeartbeatTime)
-      .map(([peerId, _]) => peerId)
-
-    inactivePeers.forEach((peerId) => lastPeerHeartbeats.delete(peerId))
-    archipelago.clearPeers(...inactivePeers)
+    for (const [peerId, lastHeartbeat] of lastPeerHeartbeats) {
+      if (lastHeartbeat < expiredHeartbeatTime) {
+        lastPeerHeartbeats.delete(peerId)
+        archipelago.onPeerRemoved(peerId)
+      }
+    }
   }, checkHeartbeatInterval)
 
   const connectSubscription = nats.subscribe('peer.*.connect')
```

Comment:
Shouldn't this be part of the `start` function?

Reply:
well, since it's a component I didn't implement a start function, I just assume start/initialiation is the same stage